components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/cache/SessionContextCache.java

```diff
@@ -178,6 +178,12 @@ private SessionContextCacheEntry getSessionFromDB(SessionContextCacheKey key) {
                     return null;
                 }
             }
+            // Add to cache if session is available in the database.
+            if (log.isDebugEnabled()) {
+                log.debug("Session corresponding to the key : " + key.getContextId() +
+                        " found in the db and is added to the cache.");
+            }
+            super.addToCache(key, cacheEntry, resolveLoginTenantDomain(getLoginTenantDomainFromContext()));
```

Comment:
Shall we add a comment here explaining why we are adding to the cache immediately. Because later in the flow, we update the values and re add to the cache. That might confuse developers because we modify the same value and makes updating here redundant.